src/pkg/packager/publish.go

```diff
@@ -0,0 +1,369 @@
+package packager
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+	v1name "github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/content/file"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/errcode"
+)
+
+// https://github.com/docker/docs/issues/8230
+const ociLayerLimit = 127
+
+// Publish publishes the package to a registry
+//
+// This is a wrapper around the oras library
+// and much of the code was adapted from the oras CLI - https://github.com/oras-project/oras/blob/main/cmd/oras/push.go
+//
+// Authentication is handled via the Docker config file created w/ `docker login`
+func (p *Packager) Publish() error {
+	p.cfg.DeployOpts.PackagePath = p.cfg.PublishOpts.PackagePath
+	if err := p.loadZarfPkg(); err != nil {
+		return fmt.Errorf("unable to load the package: %w", err)
+	}
+	spinner := message.NewProgressSpinner("")
+	defer spinner.Stop()
+
+	if p.cfg.PublishOpts.RegistryURL == "docker.io" {
+		// docker.io is commonly used, but is not a valid registry URL
```

Comment:
Why are we rewriting this to `registry-1`?

Reply:
This is due to `docker.io` not being the resolved registry URL when you perform a docker push. While `docker.io` is a popular alias, the real registry address is `registry-1`. This is my experience, but would love to be shot down (currently the publish will error due to this behavior.

Reply:
Yeah that is interesting: https://docs.docker.com/registry/introduction/

Reply:
https://github.com/docker/cli/blob/master/man/src/image/pull.md